code/obj/item/gun/gun_parent.dm

```diff
@@ -182,6 +182,34 @@ var/list/forensic_IDs = new/list() //Global list of all guns, based on bioholder
 		..()
 		ownerGun.shoot(target_turf, user_turf, owner, pox, poy)
 
+/datum/action/bar/icon/guncharge_pointblank
+	duration = 150
+	interrupt_flags = INTERRUPT_MOVE | INTERRUPT_ACT | INTERRUPT_STUNNED | INTERRUPT_ACTION
+	id = "guncharge"
+	icon = 'icons/obj/items/tools/screwdriver.dmi'
+	icon_state = "screwdriver"
+	var/obj/item/gun/ownerGun
+	var/atom/target
+	var/mob/user
+	var/second_shot
+
+	New(_gun, _target, _user, _second_shot, _time, _icon, _icon_state)
+		ownerGun = _gun
+		target = _target
+		user = _user
+		second_shot = _second_shot
+		icon = _icon
+		icon_state = _icon_state
+		duration = _time
+		..()
+
+	onEnd()
+		..()
+		var/charge_up = ownerGun.charge_up
+		ownerGun.charge_up = 0 // This is to bypass the 'charge_up' check in 'shoot_point_blank', as to prevent an actionbar loop. This is better than duplicating the entire 'shoot_point_blank' proc.
+		ownerGun.shoot_point_blank(target, user, second_shot)
+		ownerGun.charge_up = charge_up
```

Comment:
Probably cleaner to have a `skip_charge_up` argument instead. Not sure

Reply:
It seems far better than using `charge_up` itself to skip the check.

Reply:
I meant like, as an argument to the `shoot_point_blank()` proc, not as a var on the gun 😓

Reply:
Ah, my mistake; this is how I orginally coded it, using a bypass argument on the proc, but upon considering the comment about `second_shot`, I thought it would be cleaner to skip it using the method above. I'll change it back now!